src/client/modules/my/resultsFetcher/resultsFetcher.ts

```diff
@@ -43,10 +44,33 @@ export function getRawResults(): Promise<GameResultsData[]> {
     return new Promise((resolve) => resolve(cachedData));
 }
 
-export function extractPlayerStats(resultsData: GameResultsData[]): PlayerStatsAllGames[] {
+
+export function extractPlayerStats(resultsData: GameResultsData[],game_size: number): PlayerStatsAllGames[] {
     let PlayerStats: PlayerStatsAllGames[] = [];
+    // Game size array
+    let gameSize: PlayersPerGame[] = [];
+    const playersPerGame = resultsData.reduce((accum, gd: GameResultsData) => {
+        const {game_label} = gd;
+        if (!accum[game_label]) {
+            accum[game_label] = 1;
+        } else {
+            accum[game_label]++;
+        }
+        return accum;
+        
+    }, {});
+    this.gameSize = Object.entries(playersPerGame).map(([key, value]) => { return {game_label: key, player_num: value}});
+    let rawDataFiltered: GameResultsData[] = []
+    // if game size is between 2-6 filter for only games of that player size else return all data
+    if(game_size >= 2 && game_size <=6){
```

Comment:
Special pivots like this are an anti-pattern. There's no correlation between game_size=1 and the outcome. This should be a new method (with sub-methods for shared processing)